src/api/server.rkt

```diff
@@ -151,33 +141,361 @@
+  (define repr (context-repr (test-context test)))
+  (hasheq 'command
+          (get-command herbie-result)
+          'points
+          (pcontext->json pctx repr)
+          'job
+          job-id
+          'path
+          (make-path job-id)))
+
+(define (make-calculate-result herbie-result job-id)
+  (hasheq 'command
+          (get-command herbie-result)
+          'points
+          (job-result-backend herbie-result)
+          'job
+          job-id
+          'path
+          (make-path job-id)))
+
+(define (make-cost-result herbie-result job-id)
+  (hasheq 'command
+          (get-command herbie-result)
+          'cost
+          (job-result-backend herbie-result)
+          'job
+          job-id
+          'path
+          (make-path job-id)))
+
+(define (make-error-result herbie-result job-id)
+  (define errs
+    (for/list ([pt&err (job-result-backend herbie-result)])
+      (define pt (first pt&err))
+      (define err (second pt&err))
+      (list pt (format-bits (ulps->bits err)))))
+  (hasheq 'command (get-command herbie-result) 'points errs 'job job-id 'path (make-path job-id)))
+
+(define (make-exacts-result herbie-result job-id)
+  (hasheq 'command
+          (get-command herbie-result)
+          'points
+          (job-result-backend herbie-result)
+          'job
+          job-id
+          'path
+          (make-path job-id)))
+
+(define (make-improve-result herbie-result test job-id)
+  (define ctx (context->json (test-context test)))
+  (define backend (job-result-backend herbie-result))
+  (define job-time (job-result-time herbie-result))
+  (define warnings (job-result-warnings herbie-result))
+  (define timeline (job-result-timeline herbie-result))
+
+  (define repr (test-output-repr test))
+  (define backend-hash
+    (match (job-result-status herbie-result)
+      ['success (backend-improve-result-hash-table backend repr test)]
+      [_ #f]))
+
+  (hasheq 'command
+          (get-command herbie-result)
+          'status
+          (job-result-status herbie-result)
+          'test
+          test
+          'ctx
+          ctx
+          'time
+          job-time
+          'warnings
+          warnings
+          'timeline
+          timeline
+          'backend
+          backend-hash
+          'job
+          job-id
+          'path
+          (make-path job-id)))
+
+(define (backend-improve-result-hash-table backend repr test)
+  (define pcontext (improve-result-pctxs backend))
+
+  (define preprocessing (improve-result-preprocess backend))
+  (define end-hash-table (end-hash (improve-result-end backend) repr preprocessing pcontext test))
+
+  (hasheq 'preprocessing
+          preprocessing
+          'pctxs
+          pcontext
+          'start
+          (improve-result-start backend)
+          'target
+          (improve-result-target backend)
+          'end
+          end-hash-table
+          'bogosity
+          (improve-result-bogosity backend)))
```

Comment:
Why no job id and path? It might make more sense to add the job id and path in the receptionist, that way it’s uniform.

Reply:
Because `backend` is always nested in a "result-hash" (always `/improve` right now) that contains the job id and path.